app/models/host/managed.rb

```diff
@@ -560,7 +560,16 @@ def inherited_attributes
     inherited_attrs = %w{domain_id operatingsystem_id architecture_id compute_resource_id}
     inherited_attrs << "subnet_id" unless compute_provides?(:ip)
     inherited_attrs << "subnet6_id" unless compute_provides?(:ip6)
-    inherited_attrs.concat(%w{medium_id ptable_id pxe_loader}) unless image_build?
+    inherited_attrs.concat(%w{ptable_id pxe_loader}) unless image_build?
+
+    if !image_build? && hostgroup.medium_id.present?
+      if operatingsystem_id.nil? || operatingsystem.medium_ids.include?(hostgroup.medium_id)
```

Comment:
if the OS id is `nil`, does it make sense to assign a medium? I'd skip it unless it's valid
```suggestion
      if operatingsystem&.medium_ids&.include?(hostgroup.medium_id)
```

But at this point `operatingsystem` probably is `nil` because `assign_hostgroup_attributes` hasn't been called yet. It could be inherited. `medium_id` has the same issue. I think you're not really covering the real situation.

Reply:
sub-man registers the host first. So, the host was created already and it has a operating system assigned. 
The logic behind the if statement is:

If the host doesn't has a operatingsystem right now, its fine that the inherited values of operatingsystem_id and medium_id from hostgroup are taken. the operatingsystem_id is already selected here: https://github.com/theforeman/foreman/pull/10246/files#diff-b7f82bd4fe3d5a5a8b246f502ab1e894197164cf7bf48fbfd5c2f3d76da2362eR560 

If the host HAS an operatingsystem then, the hostgroup medium can ONLY be used, if the already selected operatingsystem has this medium in its medium_ids.

Reply:
> sub-man registers the host first.

But how does it register it? I'd expect as unmanaged.

Then I think we should also ask ourselves if an unmanaged host should inherit a medium at all. Once you start to manage it, wouldn't it make sense to force a medium selection then?